### PR TITLE
[Telemetry Operator] Restructure main function

### PR DESCRIPTION
**Pull request by skhalash**
**Description**

Changes proposed in this pull request:

- Split the `main` function into smaller chunks to have a better glanceability
- Move common controller metric registration in its own package

**Related issue(s)**

### CODE REVIEW

File: components/telemetry-operator/controller/logparser/controller.go [hunk 1]
```diff
@@ -20,7 +20,8 @@ import (
 	"context"
 	"fmt"
 
-	"github.com/prometheus/client_golang/prometheus"
+	commonmetrics "github.com/kyma-project/kyma/components/telemetry-operator/controller/metrics"
```
Comment (lindnerby) [inline components/telemetry-operator/controller/logparser/controller.go:23]: Why 'commonmetrics'?  Just 'metrics' also fine?

Comment (skhalash) [reply components/telemetry-operator/controller/logparser/controller.go:23]: Just to be consistent with https://github.com/kyma-project/kyma/pull/15336/files#diff-2de746570890a034850c4581b8cb422d860cb434672d4a22b54d33ab6bffae4aR69. There `metrics` is already taken.

Comment (lindnerby) [reply components/telemetry-operator/controller/logparser/controller.go:23]: Then let's name it controllermetrics

**Review by lindnerby — APPROVED** [review]

**Review by lindnerby — APPROVED** [review]

**Pull request merged**